Agents/ForecastingAgent/forecasting/forecasting_agent/agent.py

```diff
@@ -0,0 +1,526 @@
+            f'Cannot split series at {cfg["forecast_start_date"]} - out of range of series start {series.start_time()} and end {series.end_time()}')
+
+    # load the model
+    # NOTE: If you have multiple different models, you need to edit here the loading function,
+    # you can use the model_configuration_name to load the correct model or
+    # add a function to the model config like for loading the covariates
+    if 'TFT_HEAT_SUPPLY' == cfg['model_configuration_name']:
+        model = load_pretrained_model(
+            cfg, TFTModel)
+        # other models than TFT can have different key then 'input_chunk_length'
+        cfg['fc_model']['input_length'] = model.model_params['input_chunk_length']
+
+    elif 'DEFAULT' == cfg['model_configuration_name']:
+        model = Prophet()
+        cfg['fc_model']['input_length'] = len(series)
+    else:
+        raise ValueError(
+            f'No model found for model_configuration_name {cfg["model_configuration_name"]}, use one of {MODEL_MAPPING.keys()}')
+
+    forecast = get_forecast(series, covariates, model, cfg)
+
+    # create metadata
+    # input series range
+    start_date = series.end_time() - series.freq * \
+        (cfg['fc_model']['input_length'] - 1)
+    cfg['model_input_interval'] = [start_date, series.end_time()]
+    cfg['model_output_interval'] = [forecast.start_time(), forecast.end_time()]
+    cfg['created_at'] = pd.to_datetime('now', utc=True)
+    try:
+        # get unit from iri and add to forecast
+        cfg['unit'] = get_unit(cfg['iri'], kgClient)
+    except KGException as e:
+        # no measurement -> no unit
+        pass
+    cfg['time_format'] = get_time_format(cfg['iri'], kgClient)
+
+    update = get_forecast_update(cfg=cfg)
+    kgClient.performUpdate(add_insert_data(update))
+
+    # call client
+    instantiate_forecast_timeseries(tsClient, cfg, forecast)
+
+    #if backtest_series is not None:
+    #    cfg['error'] = calculate_error(backtest_series, forecast)
+    
+    # delete keys which you dont want in response, e.g. not json serializable objects
+    keys_to_delete = ['load_covariates_func',
+                      'time_delta', 'ts_data_type', 'frequency']
+    for key in keys_to_delete:
+        if key in cfg:
+            del cfg[key]
+    return cfg
+
+
+def instantiate_forecast_timeseries(tsClient, cfg, forecast):
+    tsClient.tsclient.initTimeSeries([cfg['forecast_iri']], [cfg['ts_data_type']], cfg['time_format'],
+                                     tsClient.conn)
+    ts = TSClient.create_timeseries([str(x) for x in forecast.time_index], [
+                                    cfg['forecast_iri']], [forecast.values().squeeze().tolist()])
+    tsClient.tsclient.addTimeSeriesData(ts, tsClient.conn)
+
+
+def get_forecast_start_date(forecast_start_date, tsClient, cfg):
+    if forecast_start_date is not None:
+        return pd.Timestamp(
+            isoparse(forecast_start_date)).tz_convert('UTC').tz_localize(None)
```

Comment:
What would happen if a user provides a TimeStamp which is not in UTC, but has another time zone clearly specified, e.g. `2020-01-13T00:00:00+02:00`. How would `tz_localize(None)` cope with that?

Reply:
I convert first to UTC, therefore the agent works internally with UTC, but drops time zone for proper functionally with darts and pretrained models.

Reply:
So, what would happen? ;-) Would the differently specified timezone be properly recognised and converted on your first conversion? Working internally with UTC seems sensible to me!

Reply:
So then the time zone of the instantiation is different from the start_date, but I would keep that, so that the time series inside the KG have a standardized format.